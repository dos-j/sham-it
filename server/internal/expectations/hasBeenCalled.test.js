const hasBeenCalled = require("./hasBeenCalled");

describe("unit: hasBeenCalled", () => {
  let handler;
  let requestStore;

  beforeEach(() => {
    requestStore = [];
    handler = hasBeenCalled(requestStore);
  });

  test("it should return 204 No Content if there arerequests in the requestStore", () => {
    requestStore.push({
      request: {
        method: "GET",
        pathname: "/test",
        query: { a: 1234 },
        headers: { accept: "application/json" }
      }
    });

    expect(handler()).toEqual({ status: 204 });
  });

  test("it should return 417 Expectation failed if there are no requests in the requestStore", () => {
    expect(handler()).toEqual(
      expect.objectContaining({
        status: 417
      })
    );
  });

  test("the error message should match the snapshot", () => {
    expect(handler().body).toMatchSnapshot();
  });
});
