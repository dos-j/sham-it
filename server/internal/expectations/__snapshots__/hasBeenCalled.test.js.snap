// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit: hasBeenCalled the error message should match the snapshot 1`] = `"Expected sham to have been called."`;
